lib/json_schema/validation.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require "json_schema/version_event_file_name"
+require "json_schema/versioned_schema_reader"
+require "json_schema/validator_adapter"
+require "initialize_with_config"
+
+module JsonSchema
+  class Validation
+    include InitializeWithConfig
+    required_config :schema, :schema_reader, :schema_validator, :version
+
+    def call(body:)
+      schema_validator.fully_validate(schema, body, schema_reader: schema_reader.new(version: version))
```

Comment:
What other types of validations are there? `half_ass_validate`? 😂 
Asking more out of curiosity :)

Reply:
From the validation Readme at: https://github.com/ruby-json-schema/json-schema

1. validate: returns a boolean on whether a validation attempt passes
2. validate!: throws a JSON::Schema::ValidationError with an appropriate message/trace on where the validation failed
3. fully_validate: builds an array of validation errors return when validation is complete